### PR TITLE
Fixed Package Pricing Logic

### PR DESCRIPTION
**Pull request by subratsahilgupta**
> [!IMPORTANT]
> Fixes package pricing logic in `priceService` to ensure correct rounding and package calculation, with comprehensive tests added.
> 
>   - **Behavior**:
>     - Fixes package pricing logic in `CalculateCost` and `CalculateCostWithBreakup` in `price.go`.
>     - Ensures rounding up for packages by default, with a minimum of one package charged if quantity is non-zero.
>     - Handles zero quantity edge case by returning early with tier unit amount.
>   - **Tests**:
>     - Adds `TestCalculateCostWithBreakup_PackageScenarios` and `TestCalculateCostWithBreakup_PackageRoundingModes` in `price_test.go`.
>     - Validates package pricing for various quantities and rounding modes.
>     - Ensures correct calculation of final cost, effective unit cost, and tier unit amount.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=flexprice%2Fflexprice&utm_source=github&utm_medium=referral)<sup> for 552fdf198159bbc3459bb27c681d66215d950a51. You can [customize](https://app.ellipsis.dev/flexprice/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: internal/service/price.go [hunk 1]
```diff
@@ -197,15 +197,26 @@ func (s *priceService) CalculateCost(ctx context.Context, price *price.Price, qu
 			return decimal.Zero
 		}
 
-		transformedQuantity := quantity.Div(decimal.NewFromInt(int64(price.TransformQuantity.DivideBy)))
+		// Calculate how many complete packages are needed to cover the quantity
+		packagesNeeded := quantity.Div(decimal.NewFromInt(int64(price.TransformQuantity.DivideBy)))
```
[the remark this answers is not in the document]
Comment (nkmishra1997) [reply internal/service/price.go:201]: can be ignored

**Comment by subratsahilgupta** [comment]
This error is not related to the pricing logic.
This is an issue with the event processing pipeline in the test environment, specifically:
1. Event consumer not starting properly in tests
2. Timing issues with async event processing
3. Potential issues with the in-memory event store implementation

**Comment by subratsahilgupta** [comment]
Test cases breakdown for package pricing and their scenarios:

1. **Round Up Mode with Partial Package**
   ```
   Scenario:
   - Package Size: 100 units
   - Package Price: $1.00
   - Usage: 50 units (partial package)
   Expected Result:
   - Rounds up to 1 package
   - Cost: $1.00
   ```

2. **Round Down Mode with Partial Package**
   ```
   Scenario:
   - Package Size: 100 units
   - Package Price: $1.00
   - Usage: 50 units (partial package)
   Expected Result:
   - Rounds down to 0 packages
   - Cost: $0.00
   ```

3. **Round Up Mode with Multiple Packages**
   ```
   Scenario:
   - Package Size: 100 units
   - Package Price: $1.00
   - Usage: 250 units (2.5 packages)
   Expected Result:
   - Rounds up to 3 packages
   - Cost: $3.00
   ```

4. **Round Down Mode with Multiple Packages**
   ```
   Scenario:
   - Package Size: 100 units
   - Package Price: $1.00
   - Usage: 250 units (2.5 packages)
   Expected Result:
   - Rounds down to 2 packages
   - Cost: $2.00
   ```

Additional Edge Cases Tested:
- Zero quantity (returns zero cost)
- Exact package boundaries (100, 200, 300 units)
- Just over package boundaries (101, 201 units)
- Just under package boundaries (99, 199 units)

The key difference between round up and round down:
- Round Up: Always charges for at least one package if there's any usage
- Round Down: Can charge for zero packages if usage is less than one package

**Comment by subratsahilgupta** [comment]
On it

**Comment by subratsahilgupta** [comment]
bugbot run

**Comment by nkmishra1997** [comment]
bugbot run

**Review by nkmishra1997 — APPROVED** [review]
